recipe/2024-02-terraform-providers.md

```diff
@@ -0,0 +1,401 @@
+This document  describes a proposal to support multiple providers, setup their provider configuration, which would be accessible to all recipes within an  environment.
+
+## Terms and definitions
+
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Terraform Provider | A Terraform provider is a plugin that allows Terraform to interact with and manage resources of a specific infrastructure platform or service, such as AWS, Azure, or Google Cloud. |
+
+
+## Objectives
+
+**Reference for new type recipeConfig and handling of secrets:** [Design document for Private Terraform Repository](https://github.com/radius-project/design-notes/blob/3644b754152edc97e641f537b20cf3d87a386c43/recipe/2024-01-support-private-terraform-repository.md)
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+https://github.com/radius-project/radius/issues/6539
+
+### Goals
+
+1. Enable users to use terraform recipes with multiple providers (including and outside of Azure, Kubernetes, AWS).
+2. Enable users to use terraform recipes with multiple configurations for the same provider using *alias*. 
+3. Enable provider configurations to be available for all recipies used in an environment.
+
+
+### Non goals
+1. Updates to Bicep Provider configuration.
+2. Authentication for providers hosted in private registries.
+3. Support for .terraformrc files.
+
+### User scenarios (optional)
+
+#### User story 1
+As an operator, I maintain a set of Terraform recipes for use with Radius. I have a set of provider configurations which are applicable across multiple recipes and I would like to configure them in a single centralized location for ease of maintenance.
+
+
+#### User story 2
+As an operator, I would like to manage cloud resources in different geographical regions. To enable this, I need to set up multiple configurations for the same provider for different regions and use an alias to refer to individual provider configurations. 
+
+## Design
+### Design details
+
+Reviewing popular provider configurations (GCP, Oracle, Heroku, DigitalOcean, Docker etc.) a large number of provider configurations can be setup by handling a combination of key value pairs, key value pairs in nested objects, secrets and environment variables.
+
+### Key-value pairs
+Key-value pairs will be parsed and saved to a Terraform configuration file.
```

Comment:
I think this can be a bit confusing.. Let's add more details about which key value pairs and which Terraform configuration file.

Reply:
updated